### ISSUE TITLE
insertMany with array of models not array of objects

### ISSUE THREAD

**Issue by d1manson** [Automattic/mongoose#6461]
I'm on `4.13.12`, but I think this applies forwards and backwards.

It seems you can call `insertMany` with either an array of plain objects or an array of mongoose models, however, if you provide mongoose models they are cloned inside the function with [the line](https://github.com/Automattic/mongoose/blob/master/lib/model.js#L2361):

```javascript
doc = new _this(doc)
```

This means that setting `isNew=false` further down inside the function gets applied to the clone rather than the original, and thus future `.save()`s of the original models are inserts not updates - this relates to #4099.

I propose to fix it by using something like the following (I don't know if there is a mongoose-ish way of doing the instance of check):

```javascript
 if(!(doc instanceof _this)){
     doc = new _this(doc);        
 }
```

**Comment by vkarpov15** [comment]
That should work, would you like to put in a PR?

**Comment by vkarpov15** [comment]
Closed by #6479
